src/graph/validator/MatchValidator.cpp

```diff
@@ -334,6 +334,10 @@ Status MatchValidator::buildEdgeInfo(const MatchPath *path,
 // Rewrite expression to fit semantic, check type and check used aliases.
 Status MatchValidator::validateFilter(const Expression *filter,
                                       WhereClauseContext &whereClauseCtx) {
+  auto vars = graph::ExpressionUtils::ExtractInnerVars(filter, qctx_);
+  if (!vars.empty()) {
+    return Status::SemanticError("Undefined parameter: " + vars[0]);
```

Comment:
Why print only the first parameter?

Reply:
If there are multiple undefined parameters then the user needs to edit the query multiple times to find all parameters. Why is printing them all at once not user-friendly?

Reply:
ACK.